File: scripts/examples/herd_immunity.jl
using Plots

R0 = 2.5
p = 0:0.01:1
Rv = R0 .* (1 .- p)

plot(p, Rv, label="Rv", xlabel="Vaccination Coverage (p)", ylabel="Effective R0")
hline!([1.0], label="Threshold", linestyle=:dash)
savefig("assets/herd_immunity.png")

File: scripts/examples/herd_immunity.py
import numpy as np
import matplotlib.pyplot as plt

R0 = 2.5
p = np.linspace(0, 1, 100)
Rv = R0 * (1 - p)

plt.plot(p, Rv, label="Rv")
plt.axhline(y=1.0, color="r", linestyle="--", label="Threshold")
plt.xlabel("Vaccination Coverage (p)")
plt.ylabel("Effective R0")
plt.title("Herd Immunity Threshold")
plt.legend()
plt.savefig("assets/herd_immunity_python.png")
